src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/SyntaxRewriter/TypeDeclarationCSharpSyntaxRewriter.cs

```diff
@@ -17,6 +17,14 @@ namespace Microsoft.DotNet.GenAPI.SyntaxRewriter
     /// </summary>
     public class TypeDeclarationCSharpSyntaxRewriter : CSharpSyntaxRewriter
     {
+        private readonly bool _addPartialModifier;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TypeDeclarationCSharpSyntaxRewriter"/> class, and optionally allows deciding whether to insert the partial modifier for types or not.
+        /// </summary>
+        /// <param name="addPartialModifier">Determines whether to insert the partial modifier for types or not.</param>
+        public TypeDeclarationCSharpSyntaxRewriter(bool addPartialModifier = true) => _addPartialModifier = addPartialModifier;
```

Comment:
It would be better to not set a default here and instead let the business logic explicitly provide the right setting. The partial modifier is only useful when generating code that should be compile-able and extend-able which I see very specific to GenAPI.

Reply:
Alright.